index.js

```diff
@@ -14,7 +14,7 @@ const assertPatternsInput = patterns => {
 };
 
 const toPatternsArray = patterns => {
-	patterns = [...new Set([patterns].flat())];
+	patterns = [...new Set([patterns].flat(Infinity))];
```

Comment:
Instead of flat infinity, why not `toPatternsArray(options.ignoreFiles)` then push `**/.gitignore`? Changing this will make `patterns` for `globby()` behave differently.

Reply:
Done.